### DISCUSSION TITLE
Interactive Video - Online Playback

### DISCUSSION THREAD

**Discussion by ashokreddy1208** [Sunbird-Ed/Community#142] [General Discussion]
Hi @karn-nahata @sharathkashyap @luckynvdu 

Can you please provide more information on below ticket which are part of 5.0 Sunbird-ED release.

 This a config or by default it will enable on Diksha, 
 This is a new  type of content or existing content type etc.,

https://project-sunbird.atlassian.net/browse/SB-30516

cc : @anandvarada @kameshbhr @mmanimegalai

**Comment by karn-nahata** [comment]
> Hi @karn-nahata @sharathkashyap @luckynvdu
> 
> Can you please provide more information on below ticket which are part of 5.0 Sunbird-ED release.
> 
> This a config or by default it will enable on Diksha, This is a new type of content or existing content type etc.,
> 
> https://project-sunbird.atlassian.net/browse/SB-30516
> 
> cc : @anandvarada @kameshbhr @mmanimegalai

Tagging @baljeetsingh here as well. This is a new content type.

@baljeetsingh can you help on whether this is deployed as a config or default?

**Reply by baljeetsingh** [reply]
1. Interactive would work when user is trying to create explanation content (coCreate)
2. we need to create a new category for interactive videos to work (for consumption and creation)
3. Interactive would work for newly created videos

**Reply by baljeetsingh** [reply]
CC: @karn-nahata , @coolbung

**Reply by kameshbhr** [reply]
>>we need to create a new category for interactive videos to work (for consumption and creation)

I am assuming this is a primary category. Each primary category definition has a config that defines its behavior using attributes (like supported mime types etc). For enabling interactivity, are there any attributes that need to be configured in the primary category definition?

**Reply by Xitija** [reply]
Hello @kamleshbhr sir,

There are no special attributes to be added in **this** primary category .

This primary category is being added to identify the Questions Sets played within the Interactive video which are similar to the Question Sets played by the Quml-player . (Interactive Video uses Quml-Player internally for displaying QuestionSets)

**Following for reference**
-[Feature main ticket ](https://project-sunbird.atlassian.net/browse/SB-30516)
-[Primary Category ticket ](https://project-sunbird.atlassian.net/browse/SB-30782)

Thanks.

cc: @baljeetsingh

**Reply by kameshbhr** [reply]
@Xitija As per my understanding, Sunbird doesn't inherently have any functionality driven by primary category. Primary Categories are only defined by adopters adopting Sunbird based on their domain. So I am not clear on how the new primary category is used.
From the documentation, it is clear how the player works. If the content has an attribute "interceptionPoints", which player understands and shows up questions sets within the video.
But from a creation stand point, I am not able to understand how interactive videos are created. Which editor will enable creating these? Does editor require some additional configuration to enable this? Can we add Question sets to any content (of any primary category) which is a video?
@surendrasinghs Can you please check and clarify this?

**Comment by kameshbhr** [comment]
I am summarizing my understanding based on the offline discussion and inputs from @pallakartheekreddy.

Capability to create Interactive Videos (videos embedded with questions) can be enabled for any primary category that supports video mime-types (mp4, webm). Once enabled, when a content of that primary category is getting created, the editor provides option to create and embed questions within the video.  To enable this, following are the two configurations to be done during deployment of coKreat app:

1. A portal environment variable  "dock_interactive_video_qset_category" has to be set with the primary category name of the question set that will be created and embedded into an interactive video. This can be any primary category whose object type is question set. For example the primary category could be "Practice Question Set" or "Interactive Question Set" or any such. The object type of these categories have to be question set. Note - only one primary category can be set as the question set.
2. In the primary category configuration of the content that needs to enable Interactive videos, the following attribute has to be set - "enableInteractivity": true. For example the primary categories that can enable interactive videos could be "Explanation Content", "Learning Resource" etc. Each of these have to support video mime-types. Note - there can be multiple primary categories that support interactive videos.

Additional Notes:
1. The above configurations are only to enable creation flow through coKreat sourcing app. Currently this is supported only in coKreat sourcing app and not in the "Workspace" of Sunbird-Ed portal.
2. There are no changes in any of the APIs. Existing question set and content creation APIs can be used to create content with videos embedded with question sets. The content created has to contain metadata as given in this document: https://project-sunbird.atlassian.net/wiki/spaces/PRD/pages/3193962714/Enable+Question-Sets+for+video+content 

cc: @baljeetsingh @surendrasinghs @Xitija @ashokreddy1208 @karn-nahata @luckynvdu @sharathprasad4tarento
